File: javascript/euler5.js
function solve(lim)
{
    var res = 1
    var n = 1
    while(n <= lim)
    {
        if(res%n != 0)
        {
            res *= n / gcd(res, n)
        }
        n += 1
    }
    return res
}

function gcd(a, b)
{
    if(b == 0)
    {
        return a
    }
    else if(a < b)
    {
        return gcd(b, a)
    }
    else
    {
        return gcd(b, a%b)
    }
}

console.log(solve(20))

File: python/euler5.py
from euler import timeit, gcd


@timeit
def solve(lim):
    l = [i for i in range(1, lim + 1)]
    res = 1
    for n in l:
        if res % n != 0:
            res *= n//gcd(res, n)
    return res


if __name__ == "__main__":
    print("PE #5 Ans: ", solve(20))
